app/js/lib/form_validation.js

```diff
@@ -83,7 +87,12 @@ var jQuery = require( 'jquery' ),
 		validationUrl: '',
 		sendFunction: null,
 		validate: function ( formValues ) {
-			var postData = {
+			var amountAsFloat = parseFloat( formValues.amount ),
+				postData;
+			if ( amountAsFloat === 0 || isNaN( amountAsFloat ) || !formValues.paymentType ) {
```

Comment:
You mean this would fail for `0,50`? Right, that could bite us. The correct way would be to handle the amount in a localized fashion, but that's out of the scope of this PR. Will put a comment in the extracted condition.